packages/rum-core/src/domain/assembly.spec.ts

```diff
@@ -1,37 +1,67 @@
 import type { ClocksState, RelativeTime, TimeStamp } from '@datadog/browser-core'
 import { ErrorSource, ExperimentalFeature, ONE_MINUTE, display } from '@datadog/browser-core'
+import type { Clock } from '@datadog/browser-core/test'
 import {
   mockEventBridge,
   cleanupSyntheticsWorkerValues,
   mockSyntheticsWorkerValues,
   mockExperimentalFeatures,
   setNavigatorOnLine,
   setNavigatorConnection,
+  registerCleanupTask,
+  mockClock,
 } from '@datadog/browser-core/test'
-import type { TestSetupBuilder } from '../../test'
-import { createRumSessionManagerMock, setup, createRawRumEvent } from '../../test'
+import {
+  createRumSessionManagerMock,
+  createRawRumEvent,
+  mockRumConfiguration,
+  mockUrlContexts,
+  mockActionContexts,
+  mockDisplayContext,
+  mockViewContexts,
+} from '../../test'
 import type { RumEventDomainContext } from '../domainContext.types'
 import type { RawRumActionEvent, RawRumEvent } from '../rawRumEvent.types'
 import { RumEventType } from '../rawRumEvent.types'
 import type { RumActionEvent, RumErrorEvent, RumEvent, RumResourceEvent } from '../rumEvent.types'
 import { startRumAssembly } from './assembly'
-import type { LifeCycle, RawRumEventCollectedData } from './lifeCycle'
-import { LifeCycleEventType } from './lifeCycle'
+import type { RawRumEventCollectedData } from './lifeCycle'
+import { LifeCycle, LifeCycleEventType } from './lifeCycle'
 import type { RumConfiguration } from './configuration'
 import type { ViewContext } from './contexts/viewContexts'
 import type { CommonContext } from './contexts/commonContext'
 import type { CiVisibilityContext } from './contexts/ciVisibilityContext'
+import type { RumSessionManager } from './rumSessionManager'
 
 describe('rum assembly', () => {
-  let setupBuilder: TestSetupBuilder
+  let lifeCycle: LifeCycle
+  let sessionManager: RumSessionManager
   let commonContext: CommonContext
   let serverRumEvents: RumEvent[]
-  let extraConfigurationOptions: Partial<RumConfiguration> = {}
   let findView: () => ViewContext
+  let partialRumConfig: Partial<RumConfiguration>
   let reportErrorSpy: jasmine.Spy<jasmine.Func>
   let ciVisibilityContext: { test_execution_id: string } | undefined
 
+  function setupAssemblyTest() {
+    startRumAssembly(
+      { ...mockRumConfiguration(), ...partialRumConfig },
+      lifeCycle,
+      sessionManager,
+      { ...mockViewContexts(), findView: () => findView() },
+      mockUrlContexts(),
+      mockActionContexts(),
+      mockDisplayContext(),
+      { get: () => ciVisibilityContext } as CiVisibilityContext,
+      () => commonContext,
+      reportErrorSpy
+    )
+  }
+
   beforeEach(() => {
+    lifeCycle = new LifeCycle()
+    sessionManager = createRumSessionManagerMock().setId('1234')
+    partialRumConfig = {}
     findView = () => ({
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Instead of having the test setup slitted between the `beforeEeach` and 'setupAssemblyTest' you could do something like this:
```ts

  function setupAssemblyTest(
    params = {
      sessionManager: createRumSessionManagerMock().setId('1234'),
      findView: () => ({ id: '7890', name: 'view name', startClocks: {} as ClocksState }),
      partialConfiguration: {} as RumSessionManagerMock,
      ciVisibilityContext: undefined as CiVisibilityContext | undefined,
      commonContext: {
        context: {},
        user: {},
        hasReplay: undefined,
      } as CommonContext,
    }
  ) {
    const lifeCycle = new LifeCycle()
    const reportErrorSpy = jasmine.createSpy('reportError')
    serverRumEvents = []
    const subscription = lifeCycle.subscribe(LifeCycleEventType.RUM_EVENT_COLLECTED, (serverRumEvent) => {
      serverRumEvents.push(serverRumEvent)
    })

    startRumAssembly(
      { ...mockRumConfiguration(), ...params.partialConfiguration },
      lifeCycle,
      params.sessionManager,
      { ...mockViewContexts(), findView: () => params.findView() },
      mockUrlContexts(),
      mockActionContexts(),
      mockDisplayContext(),
      { get: () => params.ciVisibilityContext } as CiVisibilityContext,
      () => params.commonContext,
      reportErrorSpy
    )

    registerCleanupTask(() => {
      subscription.unsubscribe()
      cleanupSyntheticsWorkerValues()
    })

    return { lifeCycle, reportErrorSpy, serverRumEvents }
  }
```

Then if a test need to override the config:
```ts
  setupAssemblyTest({
        partialConfiguration: {{
          beforeSend: (event) => (event.view.name = 'added'),
        }
  })
```
‏